src/progressive-profiling/ProgressiveProfiling.jsx

```diff
@@ -70,6 +71,11 @@ const ProgressiveProfiling = (props) => {
   }, [DASHBOARD_URL, registrationResponse]);
 
   useEffect(() => {
+    const queryParams = getAllPossibleQueryParams();
+    if (authenticatedUser?.userId && queryParams?.host) {
+      setEmbeddedHost(queryParams?.host);
```

Comment:
Do we need to keep a separate state? We can just simply use the query param constant directly in the events i.e
```
sendTrackEvent(
      'edx.bi.welcome.page.skip.link.clicked',
      {
        host: queryParams?.host,
      },
    );
```

Reply:
We already decided that its okay if its empty i.e user is coming from a normal flow.

Reply:
I was talking about the embedded flow to set event property before firing the event.